### PR TITLE
[DROOLS-6542] fix split statement when a String contains a single quote

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6542

### CODE REVIEW

**Review by tkobayas — APPROVED** [review]

**Review by hellowdan — APPROVED** [review]

File: drools-core/src/test/java/org/drools/core/util/StringUtilsTest.java [hunk 1]
```diff
@@ -320,6 +323,18 @@ public String toExternalForm() {
...
+        String text =
+                "System.out.println(\"'\");" +
+                "$visaApplication.setValidation( Validation.FAILED );" +
+                "drools.update($visaApplication);";
+        List<String> statements = splitStatements(text);
+        assertEquals(3, statements.size());
+        assertEquals("System.out.println(\"'\")", statements.get(0));
+        assertEquals("$visaApplication.setValidation( Validation.FAILED )", statements.get(1));
+        assertEquals("drools.update($visaApplication)", statements.get(2));
     }
```
Comment (pibizza) [inline drools-core/src/test/java/org/drools/core/util/StringUtilsTest.java:339]: I like a lot that we have at least a new unit test for a public method. 

I would suggest adding more unit tests to explore all corner cases, and possibly write an helper method to simplify checks and reduce duplicated code. Something like "stringsAre(String... expected) that checks that the number of splitted strings are correct.

I would not add any further integration test. Once we are sure that we are performing the split, verifying how the split works is for the unit test level.

**Review by pibizza — APPROVED** [review]
Comment re: adding more unit tests. This will help covering us for other possible issues.
